### PR TITLE
fix: enhance IlabModuleList parser on content change

### PR DESCRIPTION
**Pull request by JoySnow**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] No Sensitive Data in this change?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?
* [x] Need backport to `3.0_egg`? Yes, refer to [RPM/Egg Delivery](https://github.com/RedHatInsights/insights-core/blob/master/CONTRIBUTING.md#rpmegg-delivery) to open a new PR.
* [ ] Is this a backport from `master`? Yes, this is a backport of #PR-ID

### Complete Description of Additions/Changes:

 - to compatible for new version ilab cmd output format, and handle the cmd warning log lines
 - Jira: RHINENG-21787

## Summary by Sourcery

Enhance the IlabModuleList parser to handle the updated v2 table format with additional columns and pre-table warning lines, tighten parsing logic, and expand test coverage for various edge cases.

New Features:
- Support new v2 Ilab module list output by capturing pre-table warning/log lines into unparsed_lines and parsing the additional Absolute path column
- Automatically strip the leading “models/” prefix from parsed model names

Enhancements:
- Dynamically detect table headers and validate row lengths to improve parsing robustness
- Refine exception messages for empty content, missing headers, and unparsable table lines

Tests:
- Extend IlabModuleList tests to cover v2 format, unparsed_lines behavior, missing header/content scenarios, and updated exception messages

### CODE REVIEW

**Comment by JoySnow** [comment]
@xiangce  Removed the `parse_delimited_table` usage as we discussed, now the parsing process is way more compact. 
Please kindly help review the updates.

File: insights/parsers/ilab.py [hunk 1]
```diff
@@ -35,30 +36,44 @@ class IlabModuleList(CommandParser, list):
...
+                linesp = [_.strip() for _ in line.strip('|').split('|')]
+                if "Model Name" in line:
+                    headings = ['_'.join(h.lower().split()) for h in linesp]
+                elif headings and len(linesp) == len(headings):
+                    parsed_row = dict(zip(headings, linesp))
+                    # drop the leading "models/" if present
+                    if parsed_row["model_name"].startswith("models/"):
+                        parsed_row["model_name"] = parsed_row["model_name"][7:]
+                    if not parsed_row["model_name"]:
+                        raise ParseException("Unexpected parsed model line: %s" % line)
```
[the remark this answers is not in the document]
Comment (JoySnow) [reply insights/parsers/ilab.py]: @xiangce Yes, this parser is for command `ilab model list`, the model name is the key part, and usually it is not expected to be missing in a model info line. Here we check this "model name" column more strict than other columns. 
If it is missing, instead of continuing the parsing, considering to stop it here and treat it as a parsing failure. 
What do you think?
